tests/test_input.py

```diff
@@ -671,16 +643,91 @@ def test_config_parsing_bad_cv_folds():
 
     config_template_path = join(_my_dir, 'configs',
                                 'test_config_parsing.template.cfg')
-    config_path = fill_in_config_paths_for_parsing(config_template_path,
+    config_path = fill_in_config_options(config_template_path,
                                                    values_to_fill_dict,
                                                    'bad_cv_folds')
 
-    _parse_config_file(config_path)
+    yield check_config_parsing_value_error, config_path
+
+
+def test_config_parsing_invalid_option():
+
+    train_dir = join(_my_dir, 'train')
+    output_dir = join(_my_dir, 'output')
+
+    # make a simple config file that has an invalid option
+    values_to_fill_dict = {'experiment_name': 'config_parsing',
+                           'task': 'cross_validate',
+                           'train_directory': train_dir,
+                           'bad_option': 'whatever',
+                           'featuresets': "[['f1', 'f2', 'f3']]",
+                           'learners': "['LogisticRegression']",
+                           'log': output_dir,
+                           'results': output_dir,
+                           'objective': 'f1_score_macro'}
+
+    config_template_path = join(_my_dir, 'configs',
+                                'test_config_parsing.template.cfg')
+
+    config_path = fill_in_config_options(config_template_path,
+                                                   values_to_fill_dict,
+                                                   'invalid_option')
+
+    yield check_config_parsing_key_error, config_path
+
+
+def test_config_parsing_duplicate_option():
+
+    train_dir = join(_my_dir, 'train')
+    output_dir = join(_my_dir, 'output')
+
+    # make a simple config file that has an invalid option
```

Comment:
minor: comment should be updated to reflect that the problem with this config is that there is a duplicate key (that comes from the fill_in_config_options method)